test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,390 @@
+  */
+  // var twiceFoo = 2 * foo;
+
+
+
+  {
+    var bazBarFoo = baz(bar, foo); // should be 28
+
+    /* Use statements apply to the entire scope in which they are defined.  Even
+       if the use statement occurs after code which would access the symbols
+       without prefix, these accesses are still valid.
+    */
+    use modToUse;
+
+    writeln(bazBarFoo);
+  }
+
+
+  {
+    /* The symbols provided by a use statement are considered at an outer scope
+       to the symbols defined at the same level as the use itself.  Thus, if
+       another bar were defined here, and an access was attempted, the compiler
+       would find the bar at this scope, rather than modToUse.bar.
+    */
+    var bar = 4.0;
+
+    use modToUse;
+
+    writeln(bar); // Will output 4.0, not 2
+  }
+
+
+  var bar = false;
+  {
+    /* The symbols provided by a use statement are considered at an inner scope
+       to the symbols defined outside of the scope where the use statement
+       applies.  Thus, if another bar were defined outside of these curly
+       braces, and an access was attempted, the compiler would find the bar from
+       modToUse, rather than the outer bar.
+    */
+
+    use modToUse;
+    writeln(bar); // Will output 2, not false
+  }
+
+
+  {
+    /* Multiple modules may be used in the same use statement */
+    use modToUse, AnotherModule, ThirdModule;
+
+    if (a || b < 0) {
+      writeln(foo);
+    } else {
+      writeln(bar);
+    } // Will output foo (12)
+  }
+
+
+  {
+    /* And a scope may contain multiple use statements */
+    use modToUse;
+    use AnotherModule;
+
+    writeln(a && foo > 15); // outputs false
+  }
+
+
+  {
+    /* In either case, the modules used in this way are considered at the same
+       scope.  This means that if two modules each define a symbol with the same
+       name, and both modules are used at the same scope, attempts to access a
+       symbol by that name will result in a naming conflict:
+    */
+    use modToUse, Conflict;
+
+    writeln(foo); // Outputs 12
+    /* The following line would fail because both modToUse and Conflict define a
+       bar:
+    */
+    // writeln(bar);
+    writeln(other); // Outputs 5.0 + 3.0i
+  }
+
+  writeln();
+  writeln("Limiting a use");
+
+
+  {
+    /* To get around such conflicts, there are multiple strategies.  If only a
+       small number of symbols are desired from a particular module, you can
+       specify the symbols to bring in via an 'only' list.
+    */
+    use modToUse;
+    use Conflict only other;
+
+    writeln(foo); // Outputs 12
+    writeln(bar); // Outputs 2
+    writeln(other); // Outputs 5.0 + 3.0i
+  }
+
+
+  {
+    /* If every symbol other than the one which causes a conflict is desired,
+       you can specify the symbols to exclude via an 'except' list.
+    */
+    use Conflict;
+    use modToUse except bar;
+
+    writeln(foo); // Outputs 12
+    writeln(bar); // Outputs 5 after output in Conflict.bar function
```

Comment:
I didn't notice that set-up to have a paren-less function conflict with a symbol until now.  I worry a bit that exposing paren-less functions (which many users already find confusing) might just muddy the waters here without adding value.  Is there a strong reason to use it rather than simply another variable named bar?